src/ConnClient.cc

```diff
@@ -35,71 +37,89 @@
 #include <netdb.h>
 #include <netinet/tcp.h>
 
-#include "Connection.h"
+#include "ExceptionComm.h"
+#include "TCPConnection.h"
+#include "TLS.h"
+#include "TLSConnection.h"
+#include "Variables.h"
 
 using namespace comm;
 
-ConnClient::ConnClient() :
-    Connection(-1),
-    _server({"", 0})
+ConnClient::ConnClient(const Address& server_address, ConnClientConfig config) :
+    _config(std::move(config)),
```

Comment:
Re: move semantics, in cases like this where the method is "consuming" an argument, it is reasonable practice to pass that argument by value and call `std::move`:
```cpp
class Foo {
    std::string _bar;
public:
    Foo( std::string bar )
    : _bar( std::move( bar ) )
    {}
};
```
This minimizes the number of necessary copies in all cases:
```cpp
// a temporary string is created and passed as an r-value to Foo::Foo().
// The ctor argument is move constructed from the r-value, and becomes an l-value "bar" in ctor scope.
// std::move() casts bar to an r-value, which is then used to move-construct member variable "_bar".
// Total string copies: 0
Foo( std::string( "bar" ) )

// string "str" is created in caller scope, and passed to Foo::Foo() as an l-value reference.
// The ctor argument is copy-constructed from the l-value reference, and becomes an l-value in ctor scope.
// std::move() casts bar to an r-value, which is then used to move-construct member variable "_bar".
// Total string copies: 1
std::string str( "bar" );
Foo( str );
// other code uses "str"
```

Reply:
The advantage of the `pass-by-value then move` approach over passing by reference in c++11 is that rvalue arguments are moved instead of being copied (lvalues are copied in both cases). For the same case in c++17 you can also benefit from copy elision so the value is constructed in-place and moves are not even required.

I think there are clang-tidy/cppcheck rules to enforce this pattern when possible.

On the semantic side of things, by having a pass-by-value argument you're making explicit that you are not just using the argument but making a copy of it.